plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -42,24 +45,31 @@
     public final static String INDEX = ".pipelines";
     public final static String TYPE = "pipeline";
 
-    private Client client;
-    private final Injector injector;
+    private final Updater updater;
+    private final ClusterService clusterService;
+    private final TimeValue pipelineUpdateInterval;
+    private final PipelineConfigDocReader configDocReader;
+    private final Map<String, Processor.Builder> processorRegistry;
 
-    private volatile Updater updater;
-    private volatile CopyOnWriteHashMap<String, Pipeline> pipelines = new CopyOnWriteHashMap<>();
+    private volatile CopyOnWriteHashMap<String, PipelineReference> pipelines = new CopyOnWriteHashMap<>();
```

Comment:
why do you need `CopyOnWriteHashMap` here? you don't really edit the map, you only replace it.